galley/pkg/server/components/processing2.go

```diff
@@ -206,27 +213,29 @@ func (p *Processing2) Start() (err error) {
 }
 
 func (p *Processing2) disableExcludedKubeResources(m *schema.Metadata) schema.KubeResources {
+	// If service discovery is off, disable any resources that are *only* used in the context of service discovery.
 
-	// Behave in the same way as existing logic:
-	// - Builtin types are excluded by default.
-	// - If ServiceDiscovery is enabled, any built-in type should be readded.
+	// This is deprecated, but print a notice in case it's being set.
+	if len(p.args.ExcludedResourceKinds) != 0 {
+		scope.Warnf("ExcludedResourceKinds is not supported in the new processing component and will be ignored")
+	}
+
+	// Get the set of all collections used at least once in snapshot groups we care about
+	cols := make(map[collection.Name]bool)
+	for _, sg := range m.Snapshots() {
+		if p.args.EnableServiceDiscovery && sg.Name == "syntheticServiceEntry" {
+			continue
+		}
+		for _, c := range sg.Collections {
+			cols[c] = true
+		}
+	}
 
 	var result schema.KubeResources
 	for _, r := range m.KubeSource().Resources() {
-
-		if p.isKindExcluded(r.Kind) {
-			// Found a matching exclude directive for this KubeResource. Disable the resource.
+		// Disable a resource that isn't in any snapshot groups we care about
+		if !cols[r.Collection.Name] {
```

Comment:
I thought we established that this wouldn't work due to transforms?

Reply:
Discussed offline, decided that we'd just leave the old logic for now. (This is something we should eventually revisit).